src/core/events.js

```diff
@@ -1,27 +1,44 @@
 import EventEmitter from 'eventemitter2'
 import store from '../store/store'
 import * as selectors from '../store/selectors'
-import { each } from 'lodash'
+import { createValuesHashToValueSelector } from '../store/selectors/utils'
+import { mapKeys } from '../utils/func.js'
+import { subcribeByWatching } from './utils'
 
 const events = new EventEmitter()
-store.subscribe(handleEvent)
-
-const authenticated = (state) => state.globals.authenticated
-
-function handleEvent() {
-  const state = store.getState()
-  const data = selectors.captureSelector(state)
-  if (authenticated(state)) {
-    events.emit('ready')
-  }
-  each(selectors.isThereAValidCapture(state), (isValid, captureType) => {
-    if (isValid) events.emit(captureType+'Capture', data)
-  })
-  if (selectors.allCaptured(state)) {
-    events.emit('complete', data)
-  }
+
+const getState = () => store.getState()
```

Comment:
All of the following lines are quite complicated

Reckon it's worth wrapping it in a function, with a comment explaining why it's necessary?